training/Dockerfile

```diff
@@ -0,0 +1,35 @@
+# pull base image
+FROM python:3.9-slim-bookworm
+FROM condaforge/miniforge-pypy3
+
+WORKDIR /usr/src/training
+
+# set environment variables
+
+# install dependencies
+RUN mamba create --name dlp -y
+
+COPY environment.yml pyproject.toml ./
+RUN mamba run --live-stream -n dlp mamba env update --file environment.yml --prune
+RUN mamba run --live-stream -n dlp poetry install
+
+# create directory for the app user
+RUN mkdir -p /home/app
+
+# create the app user
+RUN addgroup --system app && adduser --system --group app
```

Comment:
are there benefits to creating a new app user rather than using the root user

Reply:
For security reasons, dockerfiles shouldn't be run as root and instead be run with a user with less permissions